resource.go

```diff
@@ -25,24 +25,46 @@ func resourceNameFromStruct(v interface{}) string {
 	return strings.ToLower(structs.Name(v))
 }
 
+func resourceNameFromMap(v interface{}) (string, error) {
+	if m, ok := v.(map[string]interface{}); !ok {
+		return "", errors.New("Invalid map type")
+	} else if val, ok := m["pinejs"].(string); ok && val != "" {
+		return val, nil
```

Comment:
It might be a little easier to parse this logic if you swapped this to `!ok || val == ""` and switched the returns, as ATM the logic is 'if error, report, else if no other error succeed else report other error', whereas 'if error report, else if other error report else succeed' feels a bit clearer.